dvc/fs/fsspec_wrapper.py

```diff
@@ -0,0 +1,127 @@
+import os
+import shutil
+from functools import lru_cache
+
+from dvc.progress import Tqdm
+
+from .base import BaseFileSystem
+
+
+# pylint: disable=no-member
+class FSSpecWrapper(BaseFileSystem):
+    def fs(self):
+        raise NotImplementedError
+
+    @lru_cache(512)
+    def _with_bucket(self, path):
+        if isinstance(path, self.PATH_CLS):
+            return f"{path.bucket}/{path.path}"
+        return path
+
+    def _strip_bucket(self, entry):
+        _, entry = entry.split("/", 1)
+        return entry
+
+    def _strip_buckets(self, entries, detail, prefix=None):
+        for entry in entries:
+            if detail:
+                entry = self._entry_hook(entry.copy())
+                entry["name"] = self._strip_bucket(entry["name"])
+            else:
+                entry = self._strip_bucket(
+                    f"{prefix}/{entry}" if prefix else entry
+                )
+            yield entry
+
+    def _entry_hook(self, entry):
+        """Simple hook method to be overriden when wanted to process
+        entries within info() and ls(detail=True) calls"""
+        return entry
+
+    def isdir(self, path_info):
+        if not self.exists(path_info):
+            return False
+
+        entry = self.info(path_info)
+        return entry["type"] == "directory" or (
+            entry["name"].endswith("/")
+            and entry["type"] == "file"
+            and entry["size"] == 0
+        )
```

Comment:
This is for `isdir` in `import-url` for directories, right? It makes importing empty dirs possible for adlfs and gsfs, correct? Could we just filter-out `dir/` objects in `_collect_dir` instead?

Reply:
> Could we just filter-out dir/ objects in _collect_dir instead?

We need to take care of this stuff in the download(); https://github.com/iterative/dvc/blob/92b95bae9b0bfed1321c3406ff2295c3118be028/dvc/fs/base.py#L276-L279